lib/board.rb

```diff
@@ -0,0 +1,55 @@
+require 'pry'
+
+class Board
+  attr_accessor :cells
+
+  def initialize
+    @cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def reset!
+    self.cells = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts "------------"
+    puts " #{@cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "------------"
+    puts " #{@cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "------------"
+    puts " #{@cells[6]} | #{cells[7]} | #{cells[8]} "
+    puts "------------"
+  end
+
+  def position(cell)
+    self.cells[cell.to_i - 1]
+  end
+
+  def cell_taken?(cell)
+    cell == "X" || cell == "O"
+  end
+
+  def taken?(number)
+    cell_taken?(self.cells[number.to_i - 1])
+  end
+
+  def full?
+    self.cells.all?{|cell| cell_taken?(cell)}
+  end
+
+  def turn_count
+    count = 0
+    self.cells.each do |cell|
+      count += 1 if cell_taken?(cell)
+    end
+    count
+  end
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  cells.count {|cell| cell == "O" || cell == "X"}
end
```